RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitWorkerSimple.cc

```diff
@@ -87,6 +91,10 @@ bool HGCalRecHitWorkerSimple::run(const edm::Event & evt, const HGCUncalibratedR
         HGCRecHitCollection & result)
 {
     DetId detid = uncalibRH.id();
+// don't produce rechit if detid is a ghost one
+    if((detid < GhostDetIdPosMax_ +1  and GhostDetIdPosMin_-1 < detid ) or (detid < GhostDetIdNegMax_+1  and GhostDetIdNegMin_-1 < detid ) )
```

Comment:
there is no check that the IDs are in the positive or negative eta.
As such the semantics of "Pos" and "Neg" is not significant and if the same feature persists the code can be extended to look in ranges of det Ids.